numpy/ma/core.py

```diff
@@ -4236,7 +4230,6 @@ def put(self, indices, values, mode='raise'):
         if m is nomask:
             m = getmask(values)
         else:
-            m = m.copy()
```

Comment:
Sorry, I'm not follow your question, we have not checked whether it is shared or not here. Please help me understand what you mean.